### PR TITLE
Adds Overcommit generator for RuboCop

### PR DESCRIPTION
**Pull request by coolprobn**
This PR adds Overcommit gem installation and configuration + RuboCop configuration required.

### CODE REVIEW

**Comment by coolprobn** [comment]
@abhaynikam Could you take a look into this PR when you have time and let me know if we need to fix anything here?

Thanks.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@
 * Update default node-version for GitHub Action generator and fixes PG setup issues. ([@abhaynikam][])
 * Updates Bootstrap generator for supporting Bootstrap 5 with popper.js. ([@abhaynikam][])
 
+## 0.12.0 (March 29th, 2023)
+* Adds Overcommit generator for RuboCop. ([@coolprobn][])
```
Comment (abhaynikam) [inline CHANGELOG.md:12]: can you please move this changelog entry to the unreleased section on line number 9? Usually, when I release the new gem version, I make sure to update the changelog with the release date.

File: lib/generators/boring/overcommit/pre_commit/rubocop/install/install_generator.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+require "generators/boring/overcommit/base_generator"
+
+module Boring
+  module Overcommit
+    module PreCommit
+      module Rubocop
+        class InstallGenerator < Boring::Overcommit::BaseGenerator
+          def configure_rubocop
+            unless rubocop_gem_exists?
+              say "\nrubocop gem is not installed. Please install it and run the generator again!", :red
```
Comment (abhaynikam) [inline lib/generators/boring/overcommit/pre_commit/rubocop/install/install_generator.rb:12]: can we run rubocop install generator instead here?

File: CHANGELOG.md (hunk 1 above)
Comment (coolprobn) [reply CHANGELOG.md:12]: Makes sense, I will update it.

File: lib/generators/boring/overcommit/pre_commit/rubocop/install/install_generator.rb (hunk 2 above)
Comment (coolprobn) [reply lib/generators/boring/overcommit/pre_commit/rubocop/install/install_generator.rb]: I have updated the generator to use RuboCop install generator in this case @abhaynikam

**Review by abhaynikam — APPROVED** [review]
Let's merge it

**Pull request merged**